frontend/resourceadm/pages/DeployResourcePage/DeployResourcePage.tsx

```diff
@@ -29,75 +27,44 @@ export const DeployResourcePage = ({ navigateToPageWithError }: Props) => {
-  const [loadingValidatePolicy, setLoadingValidatePolicy] = useState(false);
-  const [loadingValidateResource, setLoadingValidateResource] = useState(false);
-  const [hasError, setHasError] = useState(false);
-
   const [isLocalRepoInSync, setIsLocalRepoInSync] = useState(false);
-
-  const [hasResourceError, setHasResourceError] = useState(true);
   const [hasPolicyError, setHasPolicyError] = useState<'none' | 'validationFailed' | 'notExisting'>(
     'none'
   );
+  const [newVersionText, setNewVersionText] = useState('');
 
+  // Queries to get metadata
   const { data: repoStatus } = useRepoStatusQuery(selectedContext, repo);
-
-  const [newVersionText, setNewVersionText] = useState('');
-  const [versionInTest, setVersionInTest] = useState('');
-  const [versionInProd, setVersionInProd] = useState('');
-  const [localVersion, setLocalVersion] = useState('');
+  const { data: versionData, isLoading: versionLoading } = useResourcePolicyPublishStatusQuery(
+    selectedContext,
+    repo,
+    resourceId
+  );
+  const { data: validatePolicyData, isLoading: validatePolicyLoading } = useValidatePolicyQuery(
+    selectedContext,
+    repo,
+    resourceId
+  );
+  const { data: validateResourceData, isLoading: validateResourceLoading } =
+    useValidateResourceQuery(selectedContext, repo, resourceId);
 
   /**
-   * When the page loads, validate the resource and policy and display error / success
+   * Set the value for policy error
    */
-  useOnce(() => {
-    setLoadingPublishStatus(true);
-    setLoadingValidatePolicy(true);
-    setLoadingValidateResource(true);
-
-    get(getPublishStatusUrl(selectedContext, repo, resourceId))
-      .then((res) => {
-        const versions: ResourceVersionStatusType = res as ResourceVersionStatusType;
-
-        setVersionInTest(versions.publishedVersions.find((v) => v.environment === 'TT02').version);
-        setVersionInProd(versions.publishedVersions.find((v) => v.environment === 'PROD').version);
-        setLocalVersion(versions.resourceVersion);
-        setNewVersionText(versions.resourceVersion ?? '');
-
-        setLoadingPublishStatus(false);
-
-        // Validate policy
-        get(getValidatePolicyUrl(selectedContext, repo, resourceId))
-          .then((validatePolicyRes) => {
-            // Remove error if status is 200
-            setHasPolicyError(validatePolicyRes.status === 200 ? 'none' : 'validationFailed');
-            setLoadingValidatePolicy(false);
-          })
-          .catch((err) => {
-            // If the ploicy does not exist, set it
-            if (err.response.status === 404) setHasPolicyError('notExisting');
-            setLoadingValidatePolicy(false);
-          });
+  useEffect(() => {
+    if (!validatePolicyLoading) {
+      if (validatePolicyData === undefined) setHasPolicyError('notExisting');
+      else if (validatePolicyData.status === 400) setHasPolicyError('validationFailed');
```

Comment:
Here is a good place to check the status since you want to return a specific message for a particular error code 👍